### PR TITLE
[Feature][Connector-V2] Fake Source support produce vector data

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request
issue : #7390 
1. The Milvus connector uses column-scale as dim, so that using fake to Milvus has to configure dimensions twice.
eg:
```
 FakeSource {
      row.num = 10
      vector.dimension= 4 <---
      schema = {
           columns = [
           {
              name = book_intro
              type = float_vector
              columnScale = 4 <---
              comment = "vector"
           }
       ]
      }
  }
```

2. Currently vector support is incomplete, based on Milvus mainly test floatVector

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-milvus-e2e/src/test/resources/fake-to-milvus.conf [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+env {
+  parallelism = 1
+  job.mode = "BATCH"
+}
+
+source {
+  FakeSource {
+      row.num = 10
+      vector.dimension= 4
+      schema = {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-milvus-e2e/src/test/resources/fake-to-milvus.conf:27]: please test all new vector type.

Comment (corgy-w) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-milvus-e2e/src/test/resources/fake-to-milvus.conf:27]: get

File: seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java [hunk 2]
```diff
@@ -320,6 +325,13 @@ public static Object convertBySeaTunnelType(SeaTunnelDataType<?> fieldType, Obje
                     vector.add(Float.parseFloat(o.toString()));
                 }
                 return vector;
+            case BINARY_VECTOR:
+            case BFLOAT16_VECTOR:
+            case FLOAT16_VECTOR:
+                ByteBuffer binaryVector = (ByteBuffer) value;
+                return gson.toJsonTree(binaryVector.array());
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java:332]: Please use jackson instead.

File: seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/batch/MilvusBufferBatchWriter.java [hunk 3]
```diff
@@ -29,7 +29,8 @@
 
 import org.apache.commons.collections4.CollectionUtils;
 
-import com.alibaba.fastjson.JSONObject;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/batch/MilvusBufferBatchWriter.java:33]: ditto

File: seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java [hunk 4]
```diff
@@ -320,6 +325,13 @@ public static Object convertBySeaTunnelType(SeaTunnelDataType<?> fieldType, Obje
                     vector.add(Float.parseFloat(o.toString()));
                 }
                 return vector;
+            case BINARY_VECTOR:
+            case BFLOAT16_VECTOR:
+            case FLOAT16_VECTOR:
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java:330]: cc @Thomas-HuWei

**Review by Hisoka-X — COMMENTED** [review]
Thanks @corgy-w ! We are close to finish!

File: seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java (hunk 2 above)
Comment (corgy-w) [reply seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/convert/MilvusConvertUtils.java:332]: milvus client need gson,do need jackson before insert/update?
Snipaste_2024-08-19_18-26-29

**Review by Hisoka-X — APPROVED** [review]

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/VectorType.java [hunk 5]
```diff
@@ -30,8 +30,8 @@ public class VectorType<T> implements SeaTunnelDataType<T> {
     public static final VectorType<Map> VECTOR_SPARSE_FLOAT_TYPE =
             new VectorType<>(Map.class, SqlType.SPARSE_FLOAT_VECTOR);
 
-    public static final VectorType<Byte> VECTOR_BINARY_TYPE =
-            new VectorType<>(Byte.class, SqlType.BINARY_VECTOR);
+    public static final VectorType<ByteBuffer> VECTOR_BINARY_TYPE =
```
Comment (hailin0) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/VectorType.java:33]: Is this change compatible with older versions?

Comment (corgy-w) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/VectorType.java:33]: Currently first connector of binary vectors is implemented in this pr with no historical version problem

**Review by hailin0 — APPROVED** [review]

**Pull request merged**